src/hypothesis/internal/conjecture/data.py

```diff
@@ -34,6 +35,28 @@ class Status(IntEnum):
     INTERESTING = 3
 
 
+class StructuralTag(object):
```

Comment:
Minor: Might it be neater to define this class with `attrs`?

```python
@attr.s(slots=True)
class StructuralTag(object):
    label = attr.ib()
```

Reply:
Yeah, true. I'd need to do hash=False, cmp=False as well, but it's probably on balance still a win.

Reply:
👍 

Also makes it clear that those are intentional omissions.